src/udev/udev-rules.c

```diff
@@ -1402,9 +1402,23 @@ static bool udev_check_conflicting_tokens(UdevRuleLine *line) {
         return true;
 }
 
+static void udev_check_duplicate_tokens(UdevRuleLine *line) {
```

Comment:
Maybe, better to merge with `udev_check_conflicting_tokens()` ?

Reply:
I played with this, but the resulted code is more complicated and less readable, so I kept this approach.

Reply:
That's not strong opinion, but how about the following?
```c
        LIST_FOREACH(tokens, token, line->tokens)
                LIST_FOREACH(tokens, i, token->tokens_next) {
                        if (!tokens_eq(token, i))
                                continue;
                        if (conflicting_op(token->op, i->op)) {
                                log_line_error(...);
                                return false;
                        }
                        if (token->op == i->op) {
                                log_line_warning(...);
                                return false;
                        }
                }

        return true;
```
Or, I think multiple warnings per line is OK. So, another suggestion is
```c
        LIST_FOREACH(tokens, token, line->tokens)
                LIST_FOREACH(tokens, i, token->tokens_next) {
                        if (!tokens_eq(token, i))
                                continue;
                        if (conflicting_op(token->op, i->op))
                                log_line_error(...);
                        else if (token->op == i->op)
                                log_line_warning(...);
                }
```

Reply:
OK, changed to issue both an error about conflicting expressions and a warning about duplicate expressions.